bundles/mapping/mapwfs2/service/UserStyleService.js

```diff
@@ -0,0 +1,75 @@
+export class UserStyleService {
+    constructor () {
+        this.styles = new Map();
+        Oskari.makeObservable(this);
+        this.mutator = this.initializeMutator();
+    }
+    initializeMutator () {
+        const me = this;
+        return {
+            saveUserStyle (layerId, style) {
+                if (!layerId || !style) {
+                    return;
+                }
+                const layerStyles = me.styles.get(layerId);
+                if (layerStyles) {
+                    const index = layerStyles.findIndex(s => s.id === style.id);
+
+                    if (index !== -1) {
+                        layerStyles[index] = style;
+                    } else {
+                        layerStyles.push(style);
+                    }
+                    me.styles.set(layerId, layerStyles);
+                } else {
+                    me.styles.set(layerId, [style]);
+                }
+                me.trigger('update');
+            },
+            removeStyle (layerId, styleId) {
+                this.removeUserStyle(layerId, styleId);
+                this.removeUserStyleFromLayer(layerId, styleId);
+            },
+            removeUserStyle (layerId, styleId) {
+                if (!layerId || !styleId) {
+                    return;
+                }
+
+                const layerStyles = me.styles.get(layerId);
+                if (layerStyles) {
+                    const index = layerStyles.findIndex(s => s.id === styleId);
+                    if (index !== -1) {
+                        layerStyles.splice(index, 1);
+                        me.styles.set(layerId, layerStyles);
+                        me.trigger('update');
+                    }
+                }
+            },
+            removeUserStyleFromLayer (layerId, styleId) {
+                const layer = Oskari.getSandbox().findMapLayerFromSelectedMapLayers(layerId);
+                if (layer && layer.getCustomStyle() && layer.getCustomStyle().id === styleId) {
+                    layer.resetStyleToDefault();
+                    Oskari.getSandbox().postRequestByName('ChangeMapLayerStyleRequest', [layerId]);
+                }
+            }
+        };
```

Comment:
Remove `initializeMutator` function.
Move these to the service and use them directly when outside React Elements.

For React Element access, wrap the service with `mutatorMixin`.

Reply:
Mutator is now removed from Service and needed removeUserStyleHandler function is passed to UserStyles React component